taichi/backends/cuda/cuda_device.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+#include <vector>
+#include <set>
+
+#include "taichi/common/core.h"
+#include "taichi/backends/cuda/cuda_driver.h"
+#include "taichi/backends/cuda/cuda_context.h"
+#include "taichi/backends/device.h"
+
+TLANG_NAMESPACE_BEGIN
+
+namespace cuda {
+
+class CudaResourceBinder : public ResourceBinder {
+ public:
+  virtual ~CudaResourceBinder() override {
```

Comment:
nit: mark implementation of virtual functions with override and remove that virtual in the beggining